tests/testthat/test-install-torsten.R

```diff
@@ -0,0 +1,157 @@
+#     fixed = TRUE
+#   )
+# })
+
+test_that("install_torsten() errors if invalid version or URL", {
+  expect_error(
+    install_torsten(version = "0.89.2"),
+    "Available Torsten versions do not include 0.89.2"
+  )
+  expect_error(
+    install_torsten(release_url = "https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/torsten_v0.89.2.tar.gz"),
+    "Download of Torsten failed. Please check if the supplied release URL is valid."
+  )
+  expect_error(
+    install_torsten(release_url = "https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/0.89.2"),
+    "cmdstanr supports installing from .tar.gz archives only"
+  )
+})
+
+test_that("install_torsten() works with version and release_url", {
+  if (getRversion() < '3.5.0') {
+    dir <- tempdir()
+  } else {
+    dir <- tempdir(check = TRUE)
+  }
+
+  expect_message(
+    expect_output(
+      install_torsten(dir = dir, cores = 4,
+                      release_url = "https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/torsten_v0.89.1.tar.gz"),
+      "--- Torsten",
+      fixed = TRUE
+    ),
+    "* Finished installing Torsten",
+    fixed = TRUE
+  )
+  expect_warning(
+    expect_message(
+      expect_output(
+        install_torsten(dir = dir, cores = 4,
+                        version = "0.89.1",
+                        # the URL is intentionally invalid to test that the version has higher priority
+                        release_url = "https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/torsten_v0.89.1.tar.gz"),
+        "--- Torsten",
+        fixed = TRUE
+      ),
+      "* Finished installing Torsten",
+      fixed = TRUE
+    ),
+    "version and release_url shouldn't both be specified",
+    fixed = TRUE
+  )
+  expect_true(dir.exists(file.path(dir, "torsten_v0.89.1")))
+})
+
+test_that("Downloads respect quiet argument", {
+  if (getRversion() < '3.5.0') {
+    dir <- tempdir()
+  } else {
+    dir <- tempdir(check = TRUE)
+  }
+  version <- latest_torsten_release()
+
+##  ver_msg <- "trying URL 'https://api.github.com/repos/stan-dev/cmdstan/releases/latest'"
+  download_msg <- paste0("trying URL 'https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/",
+                         version, ".tar.gz'")
+
+  # expect_message has trouble capturing the messages from download.file
+  # so handle manually
+  install_normal <- suppressWarnings(
+    capture.output(install_torsten(dir = dir, quiet = FALSE),
+                   type = "message")
+  )
+  install_quiet <- suppressWarnings(
+    capture.output(install_torsten(dir = dir, quiet = TRUE),
+                   type = "message")
+  )
+
+##  expect_true(any(grepl(ver_msg, install_normal, fixed = TRUE)))
+  expect_true(any(grepl(download_msg, install_normal, fixed = TRUE)))
+
+##  expect_false(any(grepl(ver_msg, install_quiet, fixed = TRUE)))
+  expect_false(any(grepl(download_msg, install_quiet, fixed = TRUE)))
+})
```

Comment:
I'd vote to delete this test because it's not worth two installations to verify that we're plumbing `quiet` through correctly.

The worst case scenario of `quiet` not working properly is somehow triggering an `install_torsten` failure. That wouldn't be great, but 1) it'd be easy for the caller to work around by using the default `quiet` value (which we test) and 2) `install_torsten` is leaf functionality that wouldn't affect other things.

However, the code is simply passing `quiet` through to cmdstanr, so that sort of breakage is unlikely. The more likely thing is just that `quiet` doesn't have an affect (e.g, because we're not properly propagating it), and then the risk is just a noisier install.

(In addition, I'd like to avoid test code that matches messages from outside packages, which they are free to change without worrying about backward incompatibility.)

Reply:
Works for me. For the moment I commented it out.